crafting-dead-immerse/src/main/java/com/craftingdead/immerse/world/action/BuildActionType.java

```diff
@@ -45,17 +48,32 @@ public abstract class BuildActionType extends ItemActionType<BuildAction> {
           .map(handler -> handler.getBase().isEmpty())
           .orElse(true);
 
+  public static final BiConsumer<LivingExtension<?, ?>, BlockPos> NOTIFY_BASE_CENTER =
+      (performer, blockPos) -> LevelExtension.getOrThrow(performer.getLevel()).getLandManager()
+          .getLandOwnerAt(blockPos)
+          .map(owner -> owner instanceof LegacyBase base ? base.getBlockPos() : null)
+          .map(basePos -> performer.getLevel()
+              .getBlockEntity(basePos, ImmerseBlockEntityTypes.BASE_CENTER.get()))
+          .flatMap(base -> base)
+          .ifPresent(base -> base.playerPlacedBlock(performer, blockPos));
```

Comment:
Move the `playerPlacedBlock` method into `LandOwner` so it's not specific to `LegacyBase`.